### DISCUSSION TITLE
Fieldstate Error Not getting removed on change of required rule

### DISCUSSION THREAD

**Discussion by Ekta34** [react-hook-form/react-hook-form#8118] [Q&A]
I am using typescript in project. Initially required rule is present for Controller, but when removed the last error in  { render -> fieldState } is not getting removed due to which not able to submit form.

**Comment by Moshyfawn** [comment]
You'll have to provide a minimal reproduction codesanbox in order for us to help you debug the issue.

**Reply by Ekta34** [reply]
https://codesandbox.io/s/vigorous-pond-mw7n6o?file=/src/App.tsx 
this is minimal code for my project how i use library. 
Here we enter value for first name which change require prop for last name. But when we save, it will still give error for last name field. 

Same problem with my project.

**Comment by Moshyfawn** [comment]
To subscribe to the value change, you want to use [the `watch` form method](https://react-hook-form.com/api/useform/watch)
```diff
- const isRequired = !getValues("isRequired")
+ const isRequired = !watch("isRequired")
```

**Reply by Ekta34** [reply]
I checked with this just now, but it didn't work. 
You can check code link, it's updated.

Thanks for answering so fast! Really appreciated !

**Reply by Moshyfawn** [reply]
Alright, so I haven't found the issue yet, but I've created [this codesanbox](https://codesandbox.io/s/tender-moon-wjk6qt?file=%2Fsrc%2FApp.tsx) to check if it's the RHF issue. Doesn't look like it.

If you enter the first name, both the uncontrolled "lastName" and the controlled "middleName" fields are not required. So, the issue is somewhere in your `FormFieldWrapper` component

**Reply by Moshyfawn** [reply]
BTW, you can set `mode: "onBlur"` on the `useForm` hook instead of doing
```js
onBlur={() => trigger()}
```
to fire the validation on each input's `blur` event.

**Reply by Ekta34** [reply]
But `FormFieldWrapper` component getting required as false. you can log and check. it is setting rules to empty object as well. now field state --> error is not getting updated. when rules are empty yet error is coming for last name is an issue. 

In my opinion there is some re-rendering problem of render prop in `Controller` component.

**Reply by Moshyfawn** [reply]
It's not the RHF's problem, but rather the React's, from what I see. I might be wrong, tho.

Consider [this example](https://codesandbox.io/s/vigorous-pateu-8vz5vp?file=%2Fsrc%2FApp.tsx): you mutate the variable inside the component which doesn't get registered by React as a new prop value.

**Comment by edward-meister** [comment]
having the same problem
